packages/synapse-react-client/src/utils/hooks/useCookiePreferences.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback } from 'react'
+import { atom, useAtomValue, useSetAtom } from 'jotai'
+
+export const getCurrentCookiePreferences = () => {
+  const prefs = localStorage.getItem(COOKIES_AGREEMENT_LOCALSTORAGE_KEY)
+  const allowNone: CookiePreference = {
+    functionalAllowed: false,
+    analyticsAllowed: false,
+  }
+  return prefs != null ? (JSON.parse(prefs) as CookiePreference) : allowNone
+}
+export const COOKIES_AGREEMENT_LOCALSTORAGE_KEY =
+  'org.sagebionetworks.security.cookies.portal.preference'
+
+export type CookiePreference = {
+  functionalAllowed: boolean
+  analyticsAllowed: boolean
+}
+
+const cookiePreferencesAtom = atom<CookiePreference>(
+  getCurrentCookiePreferences(),
+)
+
+export const useCookiePreferences = (): [
+  CookiePreference,
+  (pref: CookiePreference) => void,
+] => {
+  const cookiePreferences = useAtomValue(cookiePreferencesAtom)
+  const setCookiePreferencesAtomValue = useSetAtom(cookiePreferencesAtom)
```

Comment:
Can be simplified to one hook call
```suggestion
  const [cookiePreferences, setCookiePreferencesAtomValue] = useAtom(cookiePreferencesAtom)
```